Library/Homebrew/completions/zsh.erb

```diff
@@ -44,7 +44,7 @@ __brew_completion_caching_policy() {
   (( $#tmp )) || return 0
 
   # otherwise, invalidate if latest tap index file is missing or newer than cache file
-  tmp=( ${HOMEBREW_REPOSITORY:-/usr/local/Homebrew}/Library/Taps/*/*/.git/index(om[1]N) )
+  tmp=( ${HOMEBREW_REPOSITORY:-$(brew --repository)}/Library/Taps/*/*/.git/index(om[1]N) )
```

Comment:
```suggestion
  tmp=( $(brew --repository)/Library/Taps/*/*/.git/index(om[1]N) )
```

I'd say don't even bother defaulting to the `HOMEBREW_REPOSITORY` environment variable. That's what the other completion scripts do.